### ISSUE TITLE
Better error messages when executing shell commands fails

### ISSUE THREAD

**Issue by anzhi0708** [helix-editor/helix#7509]
Hello, I've been using Helix for years. Excellent work, and thanks ;-)

It is suggested to have better error messages when executing shell commands fails. 

Specifically, if the failure is not due to an async runtime issue, the error message should not start with **"async job failed".** 

For example, when executing a shell command that is guaranteed to fail, such as ls on a non-existent path, it is more intuitive for the error message to directly come from the SHELL's output, informing the user that the path does not exist. 

However, when **"async job failed"** is used, as in the case of Helix, it can be confusing and it is unclear whether the issue is with Tokio itself, the execution of Helix, or something else. 

Especially when the terminal size is small, only "async job failed" may be visible in the error message. Why not display the exact error output from the SHELL in a pop-up window without modification?

**Comment by nkitsaini** [comment]
Agreed, even though the message moves to the pop-up, the status bar can be used to report the exit code of the process.

**Comment by Akselmo** [comment]
I was also hoping for this. Seeing the error messages just like the successful messages in a popup terminal would be great, especially if im using a build script.

This way I could just quickly copy the path from build errors in case the build files, inside helix, without context switching all the time :)

**Comment by Lyndon-Mackay** [comment]
I am going to start looking into this

**Comment by the-mikedavis** [comment]
Fixed in https://github.com/helix-editor/helix/pull/11239
